File: discover_javascript/manipulate_the_data.js
var https = require('https');

var options = {
  hostname: 'api.github.com',
  path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
  headers: {
            'User-Agent': 'Holberton_School',
	    'Authorization': 'token ' + process.env.TOKEN
	   }
}

function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
	chunks.push(chunk);
    });
    stream.on('end', () => {
	cb(chunks.join(''));
    });
}

var req = https.request(options, function(res) {
    streamToString(res, function(jsonString){
	var jsonParse=JSON.parse(jsonString);
	jsonParse.items.map(function(o, i){
	    var final=[jsonParse.items[i].full_name].join();
	    console.log(final);
	    return 0;
	});
    });
});


req.end();

File: discover_ruby/manipulate_the_data.rb
require 'httpclient'
require 'json'

def req(url, extheaders, query)
	client = HTTPClient.new
	client.get_content(url, query, extheaders)
end

_extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token fa51959c07bf48159fbaa3c6800b12efb49f217e'
}
_query=["q", "language:ruby"], ["sort", "stars"], ["order", "desc"]
_url='https://api.github.com/search/repositories?'

_gitHash=JSON.parse(req(_url, _extheaders, _query))

puts _gitHash['items'].map{ |name| name['full_name'] }.join("\n")
